### PR TITLE
initial commit for the custom Fiddler integration

### PR DESCRIPTION
**Pull request by DannyBrock**
### What does this PR do?

A brief description of the change being made with this pull request.

### Motivation

What inspired you to submit this pull request?

### Review checklist

- [ ] PR has a [meaningful title](https://github.com/DataDog/integrations-core/blob/master/CONTRIBUTING.md#pull-request-title) or PR has the `no-changelog` label attached
- [ ] Feature or bugfix has tests
- [ ] Git history is clean
- [ ] If PR impacts documentation, docs team has been notified or an issue has been opened on the [documentation repo](https://github.com/DataDog/documentation/issues/new)

### Additional Notes

Anything else we should know when reviewing?

### CODE REVIEW

**Review by estherk15 — COMMENTED** [review]
Thanks, created a card for documentation review! [DOCS-4720](https://datadoghq.atlassian.net/browse/DOCS-4720)

File: fiddler/README.md
Comment (buraizu) [inline fiddler/README.md:26]:
```suggestion
1. Edit the `fiddler.d/conf.yaml` file, in the `conf.d/` folder at the root of your Agent's configuration directory to start collecting your Fiddler performance data. See the [sample `fiddler.d/conf.yaml`][5] for all available configuration options. The `url`, `org`, and `fiddler_api_key` parameters need to be updated for the Fiddler environment you wish the integration to query. Fiddler also suggests setting the `minimum_collection_interval` setting in the `conf.yaml` file to `300` (5 minutes).
```

File: fiddler/metadata.csv
Comment (buraizu) [inline fiddler/metadata.csv:18]: Could you update the metric descriptions from lines 8 - 18 to be more specific?

File: fiddler/manifest.json
Comment (buraizu) [inline fiddler/manifest.json:11]:
```suggestion
    "description": "Fiddler provides visibility and explainability for AI model performance",
```

**Review by buraizu — CHANGES_REQUESTED** [review]
Thanks for the PR, left some minor suggestions for the README and metric descriptions.

**Comment by DannyBrock** [comment]
@buraizu Hi Bryce, I accepted the changes you suggested and made improved metric descriptions in the metadata.csv file as suggested.  Please let me know if you would like me to do anything else.

**Review by DannyBrock — COMMENTED** [review]
These files don't mean much to me so I defer to any changes you want to make here.

File: .github/CODEOWNERS
Comment (emarsha94) [inline .github/CODEOWNERS:50]: please add @DataDog/marketplace-review to the code owners file

File: fiddler/README.md
Comment (emarsha94) [inline fiddler/README.md:46]: This should be Fiddlers support

File: fiddler/assets/dashboards/fiddler_overview.json
Comment (emarsha94) [inline fiddler/assets/dashboards/fiddler_overview.json:1]: CAn you update the title of the dashboard without the dash? Ex: Fiddler Overview

**Review by emarsha94 — CHANGES_REQUESTED** [review]
Thanks for submitting the new PR! I left a couple of suggestions. In addition, can you please crop the screenshot of the Fiddler dashboard in datadog to remove the url and top of the browser so it just shows the dashboard?

File: fiddler/README.md
Comment (buraizu) [reply fiddler/README.md:4]: Optional suggestion here to remove some redundancy from this sentence and make it more succinct.

**Review by buraizu — CHANGES_REQUESTED** [review]
Thanks for implementing those changes. This review corrects a couple of minor typos and adds further detail to the acronyms used in the new metric descriptions, so let me know if any of those are inaccurate. I also left an optional style suggestion.

File: fiddler/README.md
Comment (DannyBrock) [reply fiddler/README.md:4]: I would prefer to keep it as is since this is inline with ML parlance.

File: fiddler/metadata.csv
Comment (DannyBrock) [reply fiddler/metadata.csv:17]: Again, these acronyms are well understood in this domain.  Are you ok if we leave these as acronyms?

File: fiddler/metadata.csv
Comment (DannyBrock) [reply fiddler/metadata.csv:10]: Would prefer to leave the acronym.

File: fiddler/assets/dashboards/fiddler_overview.json
Comment (DannyBrock) [reply fiddler/assets/dashboards/fiddler_overview.json:1]: @emarsha94 -- are you asking if I can change the title of the asset /images/fiddler-overview?  Or change the image itself to have the screenshot of the dashboard example be "Fiddler Overview"?

Comment (DannyBrock) [reply fiddler/assets/dashboards/fiddler_overview.json:1]: Never mind, I understand what you're asking....

File: fiddler/metadata.csv
Comment (buraizu) [reply fiddler/metadata.csv:17]: Thanks for confirming that. Users of the Datadog documentation come from a variety of backgrounds, and so I try not to second-guess what a reader may know -- it's likely that not everyone who visits the page will be active in the domain (yet). With that said, if you feel strongly on this point, it's alright to leave the acronyms as they are.

**Review by buraizu — APPROVED** [review]
Thanks again for following up. I left a response to your point about the acronyms, but will leave the decision to you. Otherwise this LGTM.

File: fiddler/metadata.csv
Comment (DannyBrock) [reply fiddler/metadata.csv:17]: I get your point.  My data scientist colleagues think it is unnecessary so I will follow their lead.

File: fiddler/manifest.json
Comment (emarsha94) [inline fiddler/manifest.json:11]: The description should be 1-2 sentences on the value of the dashboard for customers

**Review by emarsha94 — CHANGES_REQUESTED** [review]
Requested one additional change but other than that this looks good to me! However, some of the validations are failing. To resolve the errors for validation assets, can you follow the steps below? 

1. On GitHub, Navigate to your forked repository of integrations-extras
2. Select the "Sync Fork" dropdown, then click "Update Branch" to synchronizing fork with the upstream repository.

Thanks!

**Comment by DannyBrock** [comment]
Hi @emarsha94 , I made the updates you requested.  I can't do much with the description in the manifest.json as there is a validation that makes sure it isn't longer than 80 characters.  I noticed it was failing validations previously due to this error: 
 [manifest.json: contains error: {'tile': {'description': [u'Length must be between 1 and 80.']}}]
 
 I have updated it to something short and sweet, but can't go much longer as I am at 76 characters already and it is a very short sentence as is.  Please approve this now as all checks just passed.  Thanks for your help!

File: fiddler/manifest.json
Comment (emarsha94) [inline fiddler/manifest.json:11]:
```suggestion
    "description": "Gain visibility into your ML systems with the Fiddler Datadog integration",
```

File: fiddler/CHANGELOG.md
Comment (emarsha94) [inline fiddler/CHANGELOG.md:3]:
```suggestion
## 1.0.0 / 2023-02-05
```

**Review by emarsha94 — CHANGES_REQUESTED** [review]
Please remove the period from the description. Once that is done and the date is updated, good to approve!

**Review by emarsha94 — APPROVED** [review]

**Review by ofek — APPROVED** [review]

File: fiddler/manifest.json
Comment (bgoldberg122) [inline fiddler/manifest.json:52]: We have new classifier tags that we support for submitted/queried data types. Could you please add your corresponding tags?
e.g.
```"Submitted Data Type::Metrics"
"Submitted Data Type::Logs"
"Submitted Data Type::Traces"
"Submitted Data Type::Events"
"Submitted Data Type::Incidents"
"Queried Data Type::Metrics"
"Queried Data Type::Logs"
"Queried Data Type::Traces"
"Queried Data Type::Events"
"Queried Data Type::Incidents"```

**Review by bgoldberg122 — CHANGES_REQUESTED** [review]
Hi! Thanks for getting this PR ready and addressing our comments. I left one last comment, and you can delete the `fiddler/images/.gitkeep` file. Once those changes are in I can merge this and get it out for you.

**Comment by bgoldberg122** [comment]
Thanks for making the changes! It looks like one of the validations is failing though on your tests:
`FAILED tests/test_fiddler.py::test_metric_collection - AssertionError: Some metrics are collected but not asserted`

Once it's passing again, I will merge and release this.

**Comment by DannyBrock** [comment]
Hi @bgoldberg122 , thanks for helping this along.  Can you help me understand what this error means?   I am confused that this test is failing as it passed many times in the past and we didn't edit a file that would impact this test.

**Comment by bgoldberg122** [comment]
No problem! It looks like your [test assertion](https://github.com/DataDog/integrations-extras/blob/267f766bdb6303fb246965e8eb80004b8718f946/fiddler/tests/test_fiddler.py#L24-L31) is failing. Maybe the order that you're asserting is causing this error -- could you please try changing the order that you call `aggregator.assert_all_metrics_covered()` and `aggregator.assert_metrics_using_metadata(get_metadata_metrics())`?

**Comment by DannyBrock** [comment]
@bgoldberg122 - I made the suggested change, but am now seeing the failure `FAILED tests/test_fiddler.py::test_metric_collection - AssertionError: Metadata assertion errors using metadata.csv:`

Any thoughts here?  Thanks for your help.  Not sure why any of this is failing these tests now.

**Comment by bgoldberg122** [comment]
I think they were failing before too, but it is strange that it hasn't failed every CI run. 

I think the issue is actually that you are not asserting each metric before you run the `assert_all_metrics_covered` check. You should be running `aggregator.assert_metric(<metric_name>)` on each of your metrics.

**Comment by DannyBrock** [comment]
@bgoldberg122 yeah, this is odd.  These check were definitely working in the past.  I honestly don't know what these "assert" methods are doing.  Would you have 10 minutes today to talk live about this so I am clear how to resolve?

**Comment by DannyBrock** [comment]
@bgoldberg122 -- I found out part of the problem with the checks.  Apparently our demo.fiddler.ai environment, where these tests are run against, was just upgraded to a newer version of Fiddler.  This newer version returns more model metrics than the previous version -- metrics that weren't explicitly called out in metadata.csv.  I have since added these missing metrics to metadata.csv, yet now I am seeing these checks fail both locally and in github with the error below.  It doesn't seem to matter what order I call the assertion methods in either. 

Thoughts?

tests/test_fiddler.py::test_check PASSED                                 [ 33%]
tests/test_fiddler.py::test_metric_collection FAILED                     [ 66%]
tests/test_fiddler.py::test_service_check PASSED                         [100%]

=================================== FAILURES ===================================
____________________________ test_metric_collection ____________________________
tests/test_fiddler.py:30: in test_metric_collection
    aggregator.assert_all_metrics_covered()
.tox/py38/lib/python3.8/site-packages/datadog_checks/base/stubs/aggregator.py:421: in assert_all_metrics_covered
    assert condition, msg
E   AssertionError: Some metrics are collected but not asserted:
E     Asserted Metrics:
E     	- 
E     Found metrics that are not asserted:
E     	- fiddler.accuracy
E     	- fiddler.binary_cross_entropy
E     	- fiddler.data_count
E     	- fiddler.expected_callibration_error
E     	- fiddler.feature_average
E     	- fiddler.fpr
E     	- fiddler.histogram_drift
E     	- fiddler.output_average
E     	- fiddler.traffic_count
E   assert False
----------------------------- Captured stdout call -----------------------------
Here is what is passed in to check.py's constructor...

**Comment by emarsha94** [comment]
> @bgoldberg122 -- I found out part of the problem with the checks. Apparently our demo.fiddler.ai environment, where these tests are run against, was just upgraded to a newer version of Fiddler. This newer version returns more model metrics than the previous version -- metrics that weren't explicitly called out in metadata.csv. I have since added these missing metrics to metadata.csv, yet now I am seeing these checks fail both locally and in github with the error below. It doesn't seem to matter what order I call the assertion methods in either.
> 
> Thoughts?
> 
> tests/test_fiddler.py::test_check PASSED [ 33%] tests/test_fiddler.py::test_metric_collection FAILED [ 66%] tests/test_fiddler.py::test_service_check PASSED [100%]
> 
> =================================== FAILURES =================================== ____________________________ test_metric_collection ____________________________ tests/test_fiddler.py:30: in test_metric_collection aggregator.assert_all_metrics_covered() .tox/py38/lib/python3.8/site-packages/datadog_checks/base/stubs/aggregator.py:421: in assert_all_metrics_covered assert condition, msg E AssertionError: Some metrics are collected but not asserted: E Asserted Metrics: E - E Found metrics that are not asserted: E - fiddler.accuracy E - fiddler.binary_cross_entropy E - fiddler.data_count E - fiddler.expected_callibration_error E - fiddler.feature_average E - fiddler.fpr E - fiddler.histogram_drift E - fiddler.output_average E - fiddler.traffic_count E assert False ----------------------------- Captured stdout call ----------------------------- Here is what is passed in to check.py's constructor...

@DannyBrock  We believe it could be due to the order you are asserting - can you assert the one with metadata then call aggregator.assert_all_metrics_covered()?

**Comment by DannyBrock** [comment]
Yes @emarsha94 .  I've tried both ways (both orders).  I am trying again now, but I have seen the issue now with the methods called in both orders.

**Comment by DannyBrock** [comment]
Just happened again with the order of the assertion methods as requested.

`@pytest.mark.skipif(not FIDDLER_API_KEY, reason="Local testing only")
def test_metric_collection(dd_run_check, aggregator, instance):
    # type: (Callable[[AgentCheck, bool], None], AggregatorStub, Dict[str, Any]) -> None
    check = FiddlerCheck('fiddler', {}, [instance])
    dd_run_check(check)

    aggregator.assert_service_check('fiddler.can_connect', FiddlerCheck.OK)
    aggregator.assert_metrics_using_metadata(get_metadata_metrics())
    aggregator.assert_all_metrics_covered()  `

**Comment by DannyBrock** [comment]
@bgoldberg122 @emarsha94 -- Ok.  I have gotten all of the checks to pass following the recommendation above.  The one caveat is that I have commented out `assert_all_metrics_covered()`.  The reason for this is that these metrics are dynamic, based on the ML models that are in the environment these tests run against.  For example, if there are no regression-based ML models in the Fiddler environment against which these tests run, the check will not receive the metrics `fiddler.mae` nor `fiddler.r2`.  Fiddler environments are very dynamic as models come in and out of observation -- including our own https://demo.fiddler.ai environment against which these tests are running.   Thus, if you're ok with it, I'd like to leave `test_fiddler.py` as it is right now and move forward to publishing this.

**Comment by steveny91** [comment]
I think that's okay if the environment doesn't always return the same data. I think it's fine as long as there's at least some metrics that are always present. 

Non-blocking nit:
Would be still to assert all the metrics that are always coming in as you’re doing now. Then create a list of optional metrics that might appear and then assert them with at_least=0:
https://github.com/DataDog/integrations-core/blob/master/couchbase/tests/test_integration.py#L67-L68
Then you can still call aggregator.assert_all_metrics_covered() . 

It’s just sort of acts like a safety net to know that there isn’t like 1000 other metrics that are being collected without you knowing.

File: fiddler/CHANGELOG.md
Comment (steveny91) [inline fiddler/CHANGELOG.md:5]: non blocking nit:
```suggestion
* [Added] Fiddler DD integration.
```

```suggestion
* [Added] Fiddler Datadog integration.
```

**Review by steveny91 — APPROVED** [review]

**Comment by DannyBrock** [comment]
Thanks @steveny91 .  I made the changes for the optional metrics to assert_metric(at_least=0) and then added back the call to aggregator.assert_all_metrics_covered() .  The checks in github have all passed.  I also committed your change to CHANGELOG.md

**Review by bgoldberg122 — APPROVED** [review]
Looks good! Thanks for making all the changes!

**Pull request merged**
